src/hooks/screen/useChannelHook.ts

```diff
@@ -3,41 +3,45 @@ import {PostNotificationChannelList} from '../../../types/database/schema/PostNo
 
 function useChannelHook() {
   const determinePostType = (data: PostNotificationChannelList): BaseChannelItemTypeProps => {
-    const isOwningPost = data?.rawJson?.isOwnPost;
+    const isOwnSignedPost =
+      data?.rawJson?.isOwnSignedPost && data?.channelType === 'POST_NOTIFICATION';
+    const isOwnAnonymousPost =
+      data?.rawJson?.isOwnAnonymousPost && data?.channelType === 'ANON_POST_NOTIFICATION';
     const isAnonymousPost = data?.rawJson?.isAnonym;
 
     const firstComment = data?.rawJson?.comments[0];
     const hasComment = data?.rawJson?.comments.length > 0;
     const isOwningComment = firstComment?.reaction?.isOwningReaction;
     const isAnonymousComment = firstComment?.reaction?.data?.is_anonymous;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

These lines extract information about the first comment of the post. It would be beneficial to check if `data?.rawJson?.comments` exists before accessing its properties to avoid potential runtime errors.

```diff
-     const firstComment = data?.rawJson?.comments[0];
-     const hasComment = data?.rawJson?.comments.length > 0;
-     const isOwningComment = firstComment?.reaction?.isOwningReaction;
-     const isAnonymousComment = firstComment?.reaction?.data?.is_anonymous;

+     const comments = data?.rawJson?.comments;
+     const firstComment = comments ? comments[0] : undefined;
+     const hasComment = comments ? comments.length > 0 : false;
+     const isOwningComment = firstComment?.reaction?.isOwningReaction;
+     const isAnonymousComment = firstComment?.reaction?.data?.is_anonymous;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->